### ISSUE TITLE
TOR Relay

### ISSUE THREAD

**Issue by pepeEL** [SynoCommunity/spksrc#2131]
Hi
Is any chance to add TOR Relay to buils spk for our synology ?

**Comment by piwi82** [comment]
There is no chance until you provide the links to source code, dependencies, compilation informations, license, and any other stuff that could help.

**Comment by pepeEL** [comment]
Source Code
https://www.torproject.org/dist/tor-0.2.7.6.tar.gz

**Comment by cytec** [comment]
see https://github.com/SynoCommunity/spksrc/pull/1659

**Comment by pepeEL** [comment]
Yes but link with spk which you put in this #1659 not working.

**Comment by pepeEL** [comment]
Cytec can you build latest spk tor 0.2.7.6 and maybe aplha 0.2.8.1for cedarview and share me ? Please

**Comment by cytec** [comment]
@pepeEL you've tried this one? http://cytec.us/syno/test-spk/tor/tor_cedarview-5.1_0.2.5.12-1.spk what's not working?

**Comment by pepeEL** [comment]
Ok thankjs but can you build latest 0.2.7.6 ?

**Comment by pepeEL** [comment]
Ok i install version from you 0.2.5.12 but this version has no option wizard to config. When i instlled it i have no question about configuration and now only have installed in Package Center and show RUN but i have no icon in menu and on desktop to can configure it. When i try install older version 0.2.4.23 which not run , i have under instalation question to config it.

![screen](https://cloud.githubusercontent.com/assets/16692223/12917295/db1acfaa-cf38-11e5-879a-440396688370.jpg)

**Comment by cytec** [comment]
you'll have to uninstall the current one, then install mine. There is no wizzard currently see discussion on the PR itself (plans on adding some basic config stuff) so you'll have to login over ssh and modify the `torrc` file in `/usr/local/tor/var/` yourself to match your needs... currently the only thing the SPK does is install tor and set it up to run an unprotected (no password) SOCKS 5 Proxy on `0.0.0.0:9050` more info can be found in the PR

**Comment by pepeEL** [comment]
ok but when will be add wizard ?

**Comment by cytec** [comment]
@pepeEL again... check the PR... most of your questions are already discussed there... including the Wizzard and what to add and what not to add to it... even IF there will be a wizzard as of the current state of discussion it would be a verry basic one which will allow you to decide if you'll want to install Tor as Bride, exit or no-exit node see: https://github.com/SynoCommunity/spksrc/pull/1659#issuecomment-95543210 and everything else has to be done by editing the torrc file manually... 

tl;dr: there will be no full covered install wizzard, just a basic one

**Comment by nelek** [comment]
Hello

I built a Tor package for DSM : https://forum.synology.com/enu/viewtopic.php?f=190&t=114370

Bundled Tor version is 0.2.7.10. The package includes a setup wizard. For the moment I've released versions for the armadaxp, avoton, bromolow and cedarview architectures. I also have a version for northstarplus (the synology router) More architectures can be built by request.

**Comment by pepeEL** [comment]
Ok i test it but aby body know how forward all trafić to Tor ? I USE raspberry with home automatów system with ip from my home network and it worki on port 8080 and i będę forward all trafić by tor. Anybody help me please...

**Comment by nelek** [comment]
You can forward traffic by configuring your application or network to send traffic trough a SOCKS proxy using the ip and port you've configured in the wizard.

**Comment by piokuc01** [comment]
When i connect to this system by my PC and i have set in my PC web browser proxy to TOR SOCS , the page from Automation SYstem Home not open in my local network. I need forward trafffic all from raspberry....

**Comment by pepeEL** [comment]
When i connect to this system by my PC and i have set in my PC web browser proxy to TOR SOCS , the page from Automation SYstem Home not open in my local network. I need forward trafffic all from raspberry....

**Comment by nelek** [comment]
Support questions about usage of the package can be better asked here : https://forum.synology.com/enu/viewtopic.php?f=190&t=114370 Since this is not a synocommunity released package.

**Comment by pepeEL** [comment]
OKi  i write ask on forum.

**Comment by pepeEL** [comment]
Hi
I installed you version on cedarview but install ok but when i try running it i can not. I have only info "Starting tor...tor is not running"

**Comment by cytec** [comment]
@pepeEL please use the forum to discuss this as it isn't related to this issue in any way :)
@nelek you'll may open a PR for your tor spk?

**Comment by ApunktOpunkt** [comment]
Hi,

found a syno-package that is working flawlessly on my ds214play and DSM6+: http://spk.synologyitalia.com/packages/TorRelay_Intel_x86_0.2.4.23-018.spk 
For other packages check: http://spk.synologyitalia.com/index.php?fulllist=true

Cheers,
Apunkt

**Comment by nelek** [comment]
That's a really old binary. My package is working fine on DSM6, just the status reporting is broken. I'm working on a fix and bringing my package up to date to 0.2.8.12 which will be released soon.

**Comment by ward0** [comment]
@nelek  I have a DS213J: synology_armada370_213j

https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model

Can you build a me compatible Tor relay beta pls.

I'm getting 'can't run service', I already changed my user to root to see if that was the problem, but didn't help.

Thanks in advance.

**Comment by nelek** [comment]
@ward0 again, this is not the place for asking for support about my package.

Support questions about usage of the package can be better asked here : https://forum.synology.com/enu/viewtopic.php?f=190&t=114370 Since this is not a synocommunity released package.

**Comment by ward0** [comment]
@nelek, installed the version that should be working for my DSM still getting same error.. but ok i'l ask there thx

**Comment by hugoncosta** [comment]
@ApunktOpunkt, can you still run your relay? I also have a DS214Play, but I can't run it. I'm able to install it with the version you posted, but always get a "Failed to run service" error.
